src/org/ods/orchestration/usecase/BitbucketTraceabilityUseCase.groovy

```diff
@@ -0,0 +1,170 @@
+import org.ods.services.BitbucketService
+import org.ods.util.IPipelineSteps
+
+import java.text.SimpleDateFormat
+
+class BitbucketTraceabilityUseCase {
+
+    private static final String CSV_FILE = "source-code-review.csv"
+    private static final int PAGE_LIMIT = 10
+
+    private final BitbucketService bitbucketService
+    private final IPipelineSteps steps
+    private final Project project
+
+    BitbucketTraceabilityUseCase(BitbucketService bitbucketService, IPipelineSteps steps, Project project) {
+        this.steps = steps
+        this.project = project
+        this.bitbucketService = bitbucketService
+    }
+
+    /**
+     * Create a CSV file that contains the following records
+     * for every merge event into the integration branch of every ODS component:
+     * @return absolutePath of the created file
+     */
+    @SuppressWarnings(['JavaIoPackageAccess'])
+    String generateSourceCodeReviewFile() {
+        def file = new File("${steps.env.WORKSPACE}/${CSV_FILE}")
+        if (file.exists()) {
+            file.delete()
+        }
+        file.createNewFile()
+
+        def token = bitbucketService.getToken()
+        List<Map> repos = getRepositories()
+        repos.each {
+            processRepo(token, it, file)
+        }
+        return file.absolutePath
+    }
+
+    private void processRepo(String token, Map repo, File file) {
+        def nextPage = true
+        def nextPageStart = 0
+        while (nextPage) {
+            def commits = bitbucketService.getCommitsForMainBranch(token, repo.repo, PAGE_LIMIT, nextPageStart)
+            if (commits.isLastPage) {
+                nextPage = false
+            } else {
+                nextPageStart = commits.nextPageStart
+            }
+            processCommits(token, repo, commits, file)
+        }
+    }
+
+    private List<Map> getRepositories() {
+        List<Map> result = []
+        this.project.getRepositories().each {repository ->
+            result << [repo: "${project.data.metadata.id.toLowerCase()}-${repository.id}", branch: repository.branch]
+        }
+        return result
+    }
+
+    private void processCommits(String token, Map repo, Map commits, File file) {
+        commits.values.each { commit ->
+            Map mergedPR = bitbucketService.getPRforMergedCommit(token, repo.repo, commit.id)
+            // Only changes in PR and destiny integration branch
+            if (mergedPR.values
+                && mergedPR.values[0].toRef.displayId == repo.branch) {
+                def record = new Record(getDateWithFormat(commit.committerTimestamp),
+                    getAuthor(commit.author),
+                    getReviewers(mergedPR.values[0].reviewers),
+                    mergedPR.values[0].links.self[(0)].href,
+                    commit.id,
+                    repo.repo)
+                writeCSVRecord(file, record)
+            }
+        }
+    }
+
+    private void writeCSVRecord(File file, Record record) {
```

Comment:
Why CSV? Why not JSON?

Reply:
@michaelsauter the requirement was a CSV, @metmajer can give you more reasons :)

Reply:
@hrcornejo the current implementation using a CSV file as a result is only a prototype. The final solution removes the CSV file again and puts the data into a table within the SSDS document's section "2.3 Code Reviews". Please put this PR into Draft mode and use the same branch for delivering the final solution. Thanks!

Reply:
Why not use a data structure that can be marshalled to JSON? That way you can upload this and process it further if need be.

Reply:
I don't know either, but I would suggest the same, @hrcornejo.